### PR TITLE
Rework how the parties commitments are computed

### LINKED ISSUE #10628: fix commitment computation in staged_ledger

### PR DESCRIPTION
**Pull request by nholland94**
Closes #10628.

This PR removes the fee payer protocol state predicate hash from the commitment, and moves the `memo_hash` from the partial commitment to the full commitment.

### CODE REVIEW

File: src/lib/mina_base/parties.ml [hunk 1]
```diff
@@ -376,35 +376,32 @@ module Transaction_commitment = struct
 
   let typ = Snark_params.Tick.Field.typ
 
-  let create ~other_parties_hash ~protocol_state_predicate_hash ~memo_hash : t =
+  let create ~other_parties_hash : t =
     Random_oracle.hash ~init:Hash_prefix.party_with_protocol_state_predicate
-      [| protocol_state_predicate_hash; other_parties_hash; memo_hash |]
+      [| other_parties_hash |]
```
Comment (mrmr1993) [inline src/lib/mina_base/parties.ml:381]: We're double-hashing here, I feel like we could use `other_parties_hash` directly

**Review by mrmr1993 — COMMENTED** [review]
LGTM apart from the double-hashing

File: src/lib/mina_base/parties.ml (hunk 1 above)
Comment (nholland94) [reply src/lib/mina_base/parties.ml:381]: @mrmr1993 do not need to apply the hash prefix for this?

Comment (mrmr1993) [reply src/lib/mina_base/parties.ml:381]: It should be fine to use it directly, since it's already a hash

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**